shim/src/ffi.rs

```diff
@@ -1,8 +1,65 @@
 /* automatically generated by rust-bindgen 0.60.1 */
 
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct _GoString_ {
+    pub p: *const ::std::os::raw::c_char,
+    pub n: isize,
+}
+#[test]
+fn bindgen_test_layout__GoString_() {
+    assert_eq!(
+        ::std::mem::size_of::<_GoString_>(),
+        16usize,
+        concat!("Size of: ", stringify!(_GoString_))
+    );
+    assert_eq!(
+        ::std::mem::align_of::<_GoString_>(),
+        8usize,
+        concat!("Alignment of ", stringify!(_GoString_))
+    );
+    fn test_field_p() {
+        assert_eq!(
+            unsafe {
+                let uninit = ::std::mem::MaybeUninit::<_GoString_>::uninit();
+                let ptr = uninit.as_ptr();
+                ::std::ptr::addr_of!((*ptr).p) as usize - ptr as usize
+            },
+            0usize,
+            concat!(
+                "Offset of field: ",
+                stringify!(_GoString_),
+                "::",
+                stringify!(p)
+            )
+        );
+    }
+    test_field_p();
+    fn test_field_n() {
+        assert_eq!(
+            unsafe {
+                let uninit = ::std::mem::MaybeUninit::<_GoString_>::uninit();
+                let ptr = uninit.as_ptr();
+                ::std::ptr::addr_of!((*ptr).n) as usize - ptr as usize
+            },
+            8usize,
+            concat!(
+                "Offset of field: ",
+                stringify!(_GoString_),
+                "::",
+                stringify!(n)
+            )
+        );
+    }
+    test_field_n();
+}
+pub type GoString = _GoString_;
 extern "C" {
     pub fn nativeRunWithArgs(
         argc: ::std::os::raw::c_int,
         argv: *mut *mut ::std::os::raw::c_char,
     ) -> ::std::os::raw::c_uint;
 }
+extern "C" {
+    pub fn nativeRunWithTurboState(turboStateString: GoString) -> ::std::os::raw::c_uint;
+}
```

Comment:
just making sure, this file says auto-generated. is that true? Should i attempt to understand what this is?

Reply:
Yep! Autogenerated by [bindgen](https://github.com/rust-lang/rust-bindgen). Probably not worth your time to understand it.